### PR TITLE
Feature/label smoothing

### LINKED ISSUE #1087: How to do label smoothing with loss_binary_crossentropy
I've searched the Web but have been unable to figure out how to do "label smoothing" in conjunction with loss_binary_crossentropy in Keras.  I saw an example from the Python world, in which the binary_crossentropy function seems to have a parameter called "label_smoothing", but haven't found the corresponding parameter in the R API.  Does it exist, and, if not, can I get the desired effect anyway by changing the labels themselves from strict 0 or 1 to floating point numbers in between?

Thanks.

### PR DESCRIPTION
**Pull request by dfalbel**
Add label_smoothing and from_logits argument to binary crossentropy.
Fix #1087

### CODE REVIEW

File: R/losses.R [hunk 1]
```diff
@@ -112,10 +112,27 @@ loss_sparse_categorical_crossentropy <- function(y_true, y_pred) {
 }
 attr(loss_sparse_categorical_crossentropy, "py_function_name") <- "sparse_categorical_crossentropy"
 
-#' @rdname loss_mean_squared_error
+#' Computes the binary crossentropy loss.
+#' 
+#' @inheritParams loss_mean_squared_error
+#' @param from_logits Whether `y_pred` is expected to be a logits tensor. By default, 
+#'   we assume that y_pred encodes a probability distribution.
+#' @param label_smoothing numeric in between 0 and 1. If > 0 then smooth the labels.
```
Comment (skeydan) [inline R/losses.R:120]:
```suggestion
#' @param label_smoothing numeric between 0 and 1. If > 0 then smooth the labels.
```

File: R/losses.R [hunk 2]
```diff
@@ -112,10 +112,27 @@ loss_sparse_categorical_crossentropy <- function(y_true, y_pred) {
...
+#' 
+#' @seealso [loss_mean_squared_error()]
+#' 
 #' @export
-loss_binary_crossentropy <- function(y_true, y_pred) {
-  keras$losses$binary_crossentropy(y_true, y_pred)
+loss_binary_crossentropy <- function(y_true, y_pred, from_logits = FALSE, label_smoothing=0) {
```
Comment (skeydan) [inline R/losses.R:125]:
```suggestion
loss_binary_crossentropy <- function(y_true, y_pred, from_logits = FALSE, label_smoothing = 0) {
```

File: R/losses.R [hunk 3]
```diff
@@ -112,10 +112,27 @@ loss_sparse_categorical_crossentropy <- function(y_true, y_pred) {
...
+  if (tensorflow::tf_version() >="2.2") {
+    keras$losses$binary_crossentropy(y_true, y_pred, from_logits, label_smoothing)  
+  } else{
```
Comment (skeydan) [inline R/losses.R:128]:
```suggestion
  } else {
```

File: R/losses.R [hunk 4]
```diff
@@ -112,10 +112,27 @@ loss_sparse_categorical_crossentropy <- function(y_true, y_pred) {
...
+    
+    if (! ((from_logits == FALSE) && (label_smoothing == 0))) {
+      warning("from_logits and label_smoothingare ignore in TF < 2.2.")
```
Comment (skeydan) [inline R/losses.R:131]:
```suggestion
      warning("from_logits and label_smoothing are ignored in TF < 2.2.")
```
